modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -265,19 +269,28 @@ protected int processStartTag() throws Exception {
 
 		jspWriter.write("</li>");
 
-		int totalPages = (int)Math.ceil(_totalItems / _activeDelta);
+		int totalPages = (int)Math.ceil(
+			(double)_totalItems / (double)_activeDelta);
 
-		int ellipsisCount = _activePage + _ellipsisBuffer;
+		if (_ellipsisBuffer != 0) {
+			int activeIndex = _activePage - 1;
+
+			int[] totalItems = new int[totalPages];
```

Comment:
I'm a little confused by this **totalItems** name because is almost the same name as **_totalItems** field but means different, but to be honest I can't find a perfect candidate. Maybe totalPageIndexes or pageNumberItems? But the other variables (left, center and right) should follow the same pattern to be related.